### PR TITLE
Implement POST /login

### LINKED ISSUE #18: POST /login

### PR DESCRIPTION
**Pull request by delapuente**
Fix #18

### CODE REVIEW

**Comment by delapuente** [comment]
@ferjm here you have. I reordered the commits to ease the revision. Any feedback is welcome.

File: doc/API.md [hunk 1]
```diff
@@ -100,3 +101,34 @@ Failing requests may be due to the following errors:
...
+Connection: close
+{
+  "session_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjEiLCJuYW1lIjoidXNlcm5hbWUifQ.IEMuCIdMp53kiUUoBhrxv1GAPQn2L5cqhxNmCc9f_gc"
+}
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 103: Missing or malformed authentication header.
+* status code 400, errno 400: Bad request.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
```
Comment (ferjm) [inline doc/API.md:134]: Could you add the missing errors to the list after the `The currently-defined error responses are` line, please?

File: src/users_router.rs [hunk 2]
```diff
@@ -134,10 +148,61 @@ impl UsersRouter {
         }
     }
 
+    /// Returns Some pair of valid credentials if both username and password are provided or None elsewhere.
```
Comment (ferjm) [inline src/users_router.rs:151]: nit: in general, lines shorter than 80 chars, please

File: src/users_router.rs [hunk 3]
```diff
@@ -335,3 +400,131 @@ describe! setup_tests {
...
     }
 }
+
+
+describe! login_tests {
+    before_each {
+        use super::super::users_db::{UsersDb, UserBuilder};
+        use iron::prelude::Response;
+        use iron::Headers;
+        #[allow(unused_imports)]
```
Comment (ferjm) [inline src/users_router.rs]: Please, don't use this. Instead just import whatever is needed in each specific test.

File: src/users_router.rs [hunk 4]
```diff
@@ -335,3 +400,131 @@ describe! setup_tests {
...
+        use iron::headers::{Authorization, Basic};
+        use iron::status::Status;
+        use iron_test::request;
+        use iron_test::response::extract_body_to_string;
+        use rustc_serialize::Decodable;
+        use rustc_serialize::json::{self, DecodeResult};
+        #[allow(unused_imports)]
```
Comment (ferjm) [inline src/users_router.rs]: Ditto

File: src/users_router.rs [hunk 5]
```diff
@@ -335,3 +400,131 @@ describe! setup_tests {
...
+        use super::super::errors::{ErrorBody};
+
+        #[allow(dead_code)]
```
Comment (ferjm) [inline src/users_router.rs]: Ditto.

**Comment by ferjm** [comment]
Looks great! Thank you!

**Pull request merged**